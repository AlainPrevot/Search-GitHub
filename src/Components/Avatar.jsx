import StyleAvatar from './sass/Avatar.module.sass'
import PropTypes from 'prop-types'

export const Avatar = ({avatar_url}) => {
 
  return (
    <img className={StyleAvatar.avatar} src={avatar_url} alt="Avatar" />
  )
}

Avatar.propTypes = { 
  avatar_url: PropTypes.string.isRequired,
}