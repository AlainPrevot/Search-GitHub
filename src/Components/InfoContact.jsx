import {IconCompany, IconLocation, IconTwitter, IconURL} from '../Icons'
import StyleInfoContact from './sass/InfoContact.module.sass'
import PropTypes from 'prop-types'

export const InfoContact = ({location, twitter, url, company}) => {

    let user_location = location == null ? 'Not Available' : location
    let user_twitter = twitter == null ? 'Not Available' : twitter
    let user_url = url == null ? 'Not Available' : url
    let user_company = company == null ? 'Not Available' : company

  return (
    <div className={StyleInfoContact.content}>

        <IconLocation className={StyleInfoContact.content__Icon}/>
        <span>{user_location}</span>

        <IconTwitter className={StyleInfoContact.content__Icon}/>
        <span>{user_twitter}</span>

        <IconURL className={StyleInfoContact.content__Icon}/>
        <a href={user_url}>{user_url}</a>

        <IconCompany className={StyleInfoContact.content__Icon}/>
        <span>{user_company}</span>
    </div>
  )
}

InfoContact.propTypes = {
  location: PropTypes.string.isRequired,
  twitter: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  company: PropTypes.string.isRequired,
}
