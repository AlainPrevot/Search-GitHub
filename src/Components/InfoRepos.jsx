import StyleInfoRepos from './sass/InfoRepos.module.sass'

export const InfoRepos = ({repos=0, followers=0, following=0}) => {

  return (
    <div className={StyleInfoRepos.content}>

      <div>
        <span>Repos</span>
        <span>{repos}</span>
      </div>

      <div>
        <span>Folowers</span>
        <span>{followers}</span>
      </div>
      
      <div>
        <span>Following</span>
        <span>{following}</span>
      </div>
    </div>
  )
};

