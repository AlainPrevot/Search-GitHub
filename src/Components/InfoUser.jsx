import { useDate } from '../hooks/useDate'
import StyleInfoUser from './sass/InfoUser.module.sass'

export const InfoUser = ({user, name, username, bio}) => {

    let biography = bio == null ? 'This profile has no bio' : bio

    const {day, month, year} = useDate(user);
  return (
    <div className={StyleInfoUser.content}>
        <h1>{name}</h1>
        <h2>{username}</h2>
        <p>Joined {day} {month} {year}</p>  
        <p>{biography}</p>
    </div>
  )
}
