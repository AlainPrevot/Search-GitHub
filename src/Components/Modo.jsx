import StyleModo from './sass/Modo.module.sass'

export const Modo = ({onMode}) => {
  
  return (
    <div className={StyleModo.content}>

        <h1>devfinder</h1>

        <div className={StyleModo.modo}>

            <input id="modo" type="checkbox" name='my-checkbox' onClick={() => onMode()}/>
            <label className={StyleModo.dark} htmlFor="modo">DARK</label>
            <label className={StyleModo.ligth} htmlFor="modo">LIGTH</label>
        </div>
    </div>
  )
}
