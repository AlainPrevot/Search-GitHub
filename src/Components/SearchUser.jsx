import { useState } from "react"
import { IconLupa } from "../Icons";
import StyleSearchUser from "./sass/SearchUser.module.sass";
import PropTypes from 'prop-types'

export const SearchUser = ({onNewUser}) => {

    const [inputValue, setInputValue] = useState(''); //Creo un hook

    const onInputChange =  e => { {/* Esta funcion envia los datos al input cada vez que escribo */}
        setInputValue(e.target.value); //Imprimo lo qu escribo en el input
    };

    const onSubmit = e => { //Esta funcion se ejecuta cada vez que presionamos enter
        e.preventDefault(); //Hacemos que no se recargue la pagina
        if(inputValue.trim().length <= 1) return //Controlamos que el input no este vacio y eliminamos los espacios al principio y al final 

        onNewUser(inputValue.trim()); //Enviamos el resultado fuera del componente
        setInputValue(''); //Limpiamos el componente
    };


  return (
    <form className={StyleSearchUser.form} onSubmit={onSubmit}  aria-label='form'> {/* La funcion OnSubmit hace que no se recargue la pagina cada vez que buscamos un valor */}

        <IconLupa className={StyleSearchUser.form__lupa}/>

        <input 
            className={StyleSearchUser.input}
            type="text" 
            placeholder="Search GitHub username"
            value={inputValue}
            onChange={onInputChange}
        />
        
        <button className={StyleSearchUser.button} onSubmit={onSubmit}>Search</button>
    </form>
  )
};

SearchUser.propTypes = {
    onNewUser: PropTypes.func.isRequired,
}
