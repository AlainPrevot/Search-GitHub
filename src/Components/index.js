export * from './Avatar';
export * from './CardUser';
export * from './InfoContact';
export * from './InfoRepos';
export * from './InfoUser';
export * from './Modo';
export * from './SearchUser';

