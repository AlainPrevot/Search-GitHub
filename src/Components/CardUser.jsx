import { Avatar } from "./Avatar";
import { useFetchGitHub } from "./hooks/useFetchGitHub"
import { InfoContact } from "./InfoContact";
import { InfoRepos } from "./InfoRepos";
import { InfoUser } from "./InfoUser";

import './sass/CardUser.sass'

export const CardUser = ({username}) => {


  const {avatar} = useFetchGitHub(username);


  return (
    <div className="card">
        <Avatar 
          avatar_url={avatar.avatar_url}
        />
        <InfoUser
          name={avatar.name}
          username={avatar.login}
          bio={avatar.bio}
          created_at={avatar.created_at}
        />
        <InfoRepos
          repos={avatar.public_repos}
          followers={avatar.followers}
          following={avatar.following}
        />
        <InfoContact
          location={avatar.location}
          twitter={avatar.twitter_username}
          url={avatar.html_url}
          company={avatar.company}
        />
    </div>
  )
}
