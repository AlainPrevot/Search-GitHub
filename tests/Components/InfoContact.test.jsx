import { render, screen } from "@testing-library/react";
import { InfoContact } from "../../src/Components/InfoContact";

describe('Pruebas en el InfoContact', () => { 

    const location = 'CABA'
    const twitter = 'Jaime'
    const url = 'https://jaime.com/'
    const company = 'Facebook'

    test('Debe de hacer match con el snapshot', () => { 
        const {container} = render(<InfoContact location={location} twitter={twitter} url={url} company={company}/>);
        expect(container).toMatchSnapshot();
     });

    test('Debe mostrar los datos en el componente', () => { 
        render(<InfoContact location={location} twitter={twitter} url={url} company={company}/>)

        // screen.debug()

        const {href} = screen.getByRole('link')//Extraido el href del link

        //Compruebo que los textos se dibujen dentro del componente
        expect(screen.getByText('CABA')).toBeTruthy();
        expect(screen.getByText('Jaime')).toBeTruthy();
        expect(href).toBe(url);
        expect(screen.getByText('Facebook')).toBeTruthy();
     });
 });