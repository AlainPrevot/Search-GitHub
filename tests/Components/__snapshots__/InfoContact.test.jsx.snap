// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el InfoContact Debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <svg
      height="22"
      viewBox="0 0 200 260"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M85.5 13c-25.1 3.5-46.7 16.4-61.1 36.7C12.6 66.1 6.3 90.4 9 108.9c3.8 26.8 27.4 70.1 67.7 124.3C90.1 251.3 92.8 254 97 254c4.2 0 6.9-2.7 20.3-20.8 40.3-54.2 63.9-97.5 67.7-124.3 2.7-18.5-3.6-42.8-15.4-59.2-5.1-7.2-15.9-17.9-22.9-22.6-16.2-11-41.7-16.9-61.2-14.1zm25 52.3c6 3 11.4 8.5 14.6 14.7 2 3.7 2.4 6 2.4 13 0 7.3-.4 9.2-2.7 13.5-3.2 6-8.7 11.4-14.8 14.6-6.4 3.4-19.6 3.4-25.8.1-6.3-3.4-12-8.9-15-14.6-2.3-4.4-2.7-6.3-2.7-13.6 0-7 .4-9.3 2.4-13 4.2-8.1 11.5-14.3 20.1-17.1 4.9-1.6 16.2-.4 21.5 2.4z"
      />
    </svg>
    <span>
      CABA
    </span>
    <svg
      height="22"
      viewBox="0 0 280 220"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M183.3 8c-3.4.5-9.6 2.4-13.8 4.4-19.7 9.1-32.1 30.6-30.2 52.7l.6 6.6-4.7-.4c-23-2.2-47.3-10.7-66.7-23.3-10.3-6.7-23.1-17.5-29.1-24.5-2.6-3-5.5-5.5-6.5-5.5-1.1 0-2.4 1.8-3.8 5.2-8.1 20.5-4 42.6 10.9 58 5.3 5.5 5.4 5.6 2.5 5.1-1.6-.3-5.5-1.4-8.5-2.5-9.7-3.5-9-3.6-9 1.9 0 12.2 5.2 23.5 15.5 33.8 6.3 6.4 9.8 8.6 19 12.7 2.8 1.2 2.3 1.3-6.7 1.6-6.1.2-9.8.7-9.8 1.3 0 3 8.1 16 13 20.9 8.1 8.1 22.6 15 31.4 15h3l-3.4 2.5c-5.1 3.7-19.6 11-27 13.5-10.3 3.5-20.6 5.1-32.7 5.1-9 0-11.3.3-11.3 1.4 0 1.8 22.2 12.6 32.5 15.8 33.9 10.6 72.6 9.4 104.6-3.2 39.2-15.4 70.3-49.1 84.4-91.3 1.8-5.7 4.1-13.9 5-18.2 1.8-8.8 3.5-29.1 2.8-32.8-.4-1.9.3-3.1 2.9-5.2 11.6-9.6 23.4-23.3 21.5-25.2-.3-.4-4.1.5-8.4 1.9-4.3 1.4-9.5 2.9-11.6 3.2l-3.8.7 5.8-5.8C259.5 25.5 266.6 12 263 12c-.6 0-3.3 1.1-6.1 2.5-2.7 1.4-9.5 4.1-15.1 6l-10 3.3-5.5-4.4c-2.9-2.4-7.1-5.3-9.1-6.3-9-4.6-22.7-6.6-33.9-5.1z"
      />
    </svg>
    <span>
      Jaime
    </span>
    <svg
      height="18"
      viewBox="0 0 260 260"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M174.5 4.5c-3.8.9-11.3 3.6-16.5 6.2-9.1 4.4-10.4 5.5-29.9 24.7-35 34.5-41.9 43.1-47.1 58.2-6.2 18.1-3.7 41.4 6.2 58.2 4.2 7.1 14.6 18.2 19.3 20.8l3.7 1.9 6.8-7.2c3.8-4 8.6-9.7 10.6-12.7l3.6-5.5-7.5-7.3c-6.9-6.7-7.7-7.9-9.7-14.7-1.7-5.4-2.1-8.7-1.7-12.8 1.1-10.5 3.9-14.3 31.3-42.7 30.4-31.4 32.6-33 46.4-33.1 7.3 0 9.3.4 14.2 2.8 7.2 3.5 14 11 16.8 18.5 1.8 4.6 2.1 7.2 1.8 13.2-.7 10.9-3.6 15.8-18 30.7-9 9.3-11.8 12.8-11.3 14.1 3.3 8.7 5.2 28 3.7 38.2-.5 3.6-.7 6.7-.5 7 1 1 45.3-45.5 49.7-52 7.9-11.8 11.6-24.4 11.6-38.9 0-11-2.3-20.3-7.6-31.1-8.1-16.3-23-28.9-40.9-34.6-10.3-3.2-24.9-4-35-1.9z"
      />
      <path
        d="M141.1 95.8c-3.6 3.7-7.9 9.1-9.5 12.1l-2.8 5.3 6.5 5.9c14.4 13 16.7 30.4 5.9 44.8-2.1 2.8-14.5 16.1-27.6 29.5-28.1 29.1-29.6 30.1-44.1 30.1-7.8 0-9.7-.4-14.7-2.8-14.6-7.1-21.7-23.8-16.9-39.5 2-6.7 2.9-7.9 14.9-20.8 8.4-9 12.7-14.4 12.6-15.8 0-1.2-.9-5.9-1.9-10.5-1.3-6.3-1.6-11.5-1.3-21.8l.4-13.5-21.8 22.4C18.3 144.3 10.7 153.7 7 163.7c-13 34.5 1.3 71.9 33.5 87.8 9.4 4.6 19 6.7 30.5 6.7 15.2 0 30.4-5 41.5-13.7 5.7-4.4 49.9-49.4 55.7-56.5 8.6-10.7 13.8-26.3 13.8-41.2 0-20.7-7.3-37.4-22.4-51.6-8.8-8.2-10-8.1-18.5.6z"
      />
    </svg>
    <a
      href="https://jaime.com/"
    >
      https://jaime.com/
    </a>
    <svg
      height="22"
      viewBox="0 0 280 260"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M47.4 9.8c-3.9.6-7.6 5.6-8.2 11.1-.4 4.3-.8 5.1-3.3 6.1-6.1 2.6-5.9.7-5.9 60.8 0 48.9.2 55.1 1.6 56.3 1.1.9 2 1 3.2.2 1.5-.9 1.7-6 2.2-55.1.3-29.7.8-54.4 1.2-54.8 1.1-1.1 118.9 8.4 120.5 9.7 1.1.9 1.3 17 1.3 90V223h-9.9l-.3-28.1-.3-28.1-2.4-1.9c-2.1-1.7-4.1-1.9-21.1-1.9-25.2 0-24-.9-24 18.4 0 14.6.5 16.6 4.2 16.6 3 0 3.8-3.4 3.8-15.4V171h32v52h-32v-6.3c0-6.7-1-8.7-4.6-8.7-2.8 0-3.4 1.7-3.4 9.1v5.9H36.9l.1-33.1c.1-20-.2-33.5-.8-34.1-.5-.5-2-.8-3.3-.6l-2.4.3-.6 33.5c-.3 18.4-.7 33.6-.8 33.7 0 .2-1 .3-2.1.3-3.9 0-9 2.9-11 6.2-4.3 7.1-1.5 15.3 6.5 18.3 4.4 1.7 11 1.8 121.6 1.2 128.3-.7 121.1-.3 124.4-6.7 2-4 1.9-8.9-.5-12.7-2.3-3.8-4.3-5-9.9-5.9l-4.1-.7-.2-32.1-.3-32.1h-6l-.3 32.2-.2 32.3h-80V122h79l.5 2.2c.3 1.3.6 6.8.8 12.2.2 8 .5 10.1 1.9 10.9 1.2.8 2.1.7 3.2-.2 1.3-1.1 1.6-3.7 1.6-13.8 0-6.8-.5-13.3-1-14.4-2.6-4.7-4.3-4.9-46.5-4.9H167V78.1c0-35.1 0-35.9-2.1-38.5-1.2-1.5-3.2-3.2-4.5-3.6-2-.8-2.4-1.7-2.4-5 0-6-2.3-10.7-6.2-12.5-2.3-1-17.6-2.5-49.3-5C77.2 11.6 54.9 9.9 53 9.7c-1.9-.2-4.5-.1-5.6.1z"
      />
      <path
        d="M48.2 59.6C46.2 61 46 62 46 69.9c0 11.8.4 12.1 13.8 12.1 13 0 13.2-.2 13.2-12 0-11.4-.7-12-13.5-12-6.6 0-9.7.4-11.3 1.6zM86.2 59.6C84.2 61 84 62 84 70c0 11.8.2 12 13.4 12 13.2 0 13.6-.3 13.6-12.3 0-11.1-.7-11.7-13.5-11.7-6.6 0-9.7.4-11.3 1.6zM123 60.7c-1.9 2.5-2.1 3.8-1.8 10.6.4 10.3 1 10.7 14.3 10.7 13 0 13.5-.4 13.5-11.8 0-11.5-.8-12.2-13.9-12.2-9.7 0-10 .1-12.1 2.7zM47.7 93.8c-1.3 1.4-1.7 3.9-1.7 10.5 0 11.2.6 11.7 14.1 11.7 12.4 0 12.9-.5 12.9-12.2C73 92.4 72.5 92 59.3 92c-8.4 0-10.3.3-11.6 1.8zM85.6 93.6c-1.2 1.2-1.6 3.7-1.6 10.3 0 11.7.5 12.1 13.1 12.1 13.3 0 13.9-.5 13.9-11.9 0-11.8-.4-12.1-13.8-12.1-7.8 0-10.4.4-11.6 1.6zM123.5 93.4c-1.6 1.2-2 2.9-2.3 9.3-.3 6.8-.1 8.1 1.8 10.5 2.1 2.7 2.4 2.8 12.1 2.8 13.1 0 13.9-.7 13.9-12.3 0-11.3-.5-11.7-13.6-11.7-7.1 0-10.6.4-11.9 1.4zM47.6 127.6c-1.2 1.2-1.6 3.7-1.6 10.4 0 11.4.7 12 13.5 12s13.5-.6 13.5-11.6c0-12.5.1-12.4-13-12.4-8.5 0-11.2.3-12.4 1.6zM85.2 127.2c-.8.8-1.2 4.6-1.2 10.5 0 12.1.7 12.6 14.6 12.1 12-.4 12.4-.8 12.4-12.4 0-11.2-.2-11.4-13.9-11.4-7 0-11.1.4-11.9 1.2zM122.9 127.7c-.8.9-1.5 4.8-1.8 9.4-.3 7.3-.1 7.9 2.3 10.4 2.6 2.6 3 2.7 12.7 2.3 12.5-.4 12.9-.8 12.9-12.3 0-11.1-.5-11.5-14.2-11.5-8.6 0-10.8.3-11.9 1.7zM178 141.8c-3.8 3.1-4.1 18-.4 21.7 1.2 1.2 4 1.5 12.4 1.5 13.6 0 14-.4 14-12.7 0-11.6-.8-12.3-13.9-12.3-8.1 0-10.3.3-12.1 1.8zM212 142.7c-1.9 2.5-2.1 3.8-1.8 11.1.4 10.9.8 11.2 14.9 11.2 13.4 0 13.9-.4 13.9-12.5 0-7.2-.3-8.4-2.3-10.3-2.1-2-3.4-2.2-12.4-2.2-9.9 0-10.2.1-12.3 2.7zM47.7 165.8c-1.3 1.4-1.7 3.8-1.7 10 0 12.5-.5 12.2 23.5 12.2s23.5.2 23.5-12c0-12.3.5-12-23.8-12-18 0-20 .2-21.5 1.8zM178.4 175.8c-2.3 1.5-3.4 5.3-3.4 12 0 5.5.4 7.5 2.1 9.6 2 2.5 2.4 2.6 12.5 2.6 13.7 0 14.4-.6 14.4-12.3 0-12.3-.5-12.7-13.8-12.7-5.9 0-11.2.4-11.8.8zM212.4 176.4c-1.6 1.2-2 2.9-2.2 9.8-.3 7.3-.1 8.6 1.8 11 2.1 2.7 2.4 2.8 12.5 2.8 13.6 0 14.5-.8 14.5-12.5 0-12.1-.5-12.5-13.9-12.5-7.7 0-11.3.4-12.7 1.4z"
      />
    </svg>
    <span>
      Facebook
    </span>
  </div>
</div>
`;
